### PR TITLE
Standard task runes

### PR DESCRIPTION
**Pull request by eccles**
Also ensure that go.mod is up to date

Test results:

     task test:unit PASS
     task azurite:preflight && task test:azurite PASS

Linked to avid:

    smoke: RESULTS: Tests 131 , Errors 0 , Failures 0 , Time 97.299
    core: RESULTS: Tests 556 , Errors 0 , Failures 0 , Time 293.290

[AB#9472](https://dev.azure.com/jitsuin/0629f48c-3979-4bbc-9026-cb06b3dfd0ae/_workitems/edit/9472)

### CODE REVIEW

**Review by j-hartley — CHANGES_REQUESTED** [review]

File: taskfiles/Taskfile_codeqa.yml [hunk 1]
```diff
@@ -24,22 +24,36 @@ vars:
 tasks:
 
   format:
-    desc: Quality assurance of code
-    summary: "format sources (go fmt)"
+    summary: Quality assurance of code
     cmds:
-      - gofmt -l -s -w .
+      - |
```
[the remark this answers is not in the document]
Comment (eccles) [reply taskfiles/Taskfile_codeqa.yml:29]: fixed

File: taskfiles/Taskfile_codeqa.yml [hunk 2]
```diff
@@ -24,22 +24,36 @@ vars:
...
-        goimports {{.VERBOSE}} -w .
+        which golangci-lint
+        golangci-lint --version
         golangci-lint {{.VERBOSE}} run --timeout 10m ./...
-        gofmt -l -s -w .
+
+  modules:
+    summary: Upgrade modules
+    cmds:
+      - |
```
[the remark this answers is not in the document]
Comment (eccles) [reply taskfiles/Taskfile_codeqa.yml:52]: fixed

**Review by robinbryce — CHANGES_REQUESTED** [review]
I don't think we want to resolve the version of grpc-gatway this way

File: taskfiles/Taskfile_codeqa.yml [hunk 3]
```diff
@@ -24,22 +24,40 @@ vars:
...
+        set -euo pipefail
+        GATEWAYVERSION=$(protoc-gen-grpc-gateway -version | cut -d',' -f1 | cut -d' ' -f2)
```
[the remark this answers is not in the document]
Comment (robinbryce) [reply taskfiles/Taskfile_codeqa.yml:54]: If this is controversial, I suggest you move this change to a different PR so it can be addressed in isolation

**Review by robinbryce — APPROVED** [review]
LGTM

**Pull request merged**